packages/library/src/rpc-subscription-coalescer.ts

```diff
@@ -0,0 +1,132 @@
+                    Parameters<PendingRpcSubscription<unknown>['subscribe']>,
+                    AsyncIterable<unknown>
+                >({
+                    getAbortSignalFromInputArgs: ({ abortSignal }) => abortSignal,
+                    getCacheEntryMissingError(deduplicationKey) {
+                        // TODO: Coded error.
+                        return new Error(
+                            `Found no cache entry for subscription with deduplication key \`${deduplicationKey?.toString()}\``
+                        );
+                    },
+                    getCacheKeyFromInputArgs: () => deduplicationKey,
+                    async onCacheHit(_iterable, _config) {
+                        /**
+                         * This transport's goal is to prevent duplicate subscriptions from
+                         * being made. If a cached iterable] is found, do not send the subscribe
+                         * message again.
+                         */
+                    },
+                    async onCreateIterable(abortSignal, config) {
+                        const pendingSubscription = (subscriptionMethod as CallableFunction)(
+                            ...rawParams
+                        ) as PendingRpcSubscription<unknown>;
+                        const iterable = await pendingSubscription.subscribe({
+                            ...config,
+                            abortSignal,
+                        });
+                        registerIterableCleanup(iterable, () => {
+                            cache.delete(deduplicationKey);
+                        });
+                        return iterable;
+                    },
+                });
+                const pendingSubscription: PendingRpcSubscription<unknown> = {
+                    async subscribe(...args) {
+                        const iterable = await iterableFactory(...args);
+                        const { abortSignal } = args[0];
+                        let abortPromise;
+                        return {
+                            ...iterable,
+                            async *[Symbol.asyncIterator]() {
+                                abortPromise ||= abortSignal.aborted
+                                    ? Promise.reject(EXPLICIT_ABORT_TOKEN)
+                                    : new Promise<never>((_, reject) => {
+                                          abortSignal.addEventListener('abort', () => {
+                                              reject(EXPLICIT_ABORT_TOKEN);
+                                          });
+                                      });
+                                try {
+                                    const iterator = iterable[Symbol.asyncIterator]();
+                                    while (true) {
+                                        const iteratorResult = await Promise.race([iterator.next(), abortPromise]);
```

Comment:
This whole thing is mega intense, but it all boils down to this.

1. Create me a new iterable distinct from all the other ones that share this subscription.
2. Make it so that _each_ one is abortable separately.
3. Every pass through the iterator, race the iterator and the abort promise; If the user aborts the subscription, it will return _right away_ without waiting for the next message from the subscription.

Reply:
Thanks, this was a help actually 😅